core/src/main/java/io/kestra/core/runners/RunContextLogger.java

```diff
@@ -277,15 +278,26 @@ protected ILoggingEvent transform(ILoggingEvent event) {
             try {
                 String message = replaceSecret(event.getMessage());
                 Object[] argumentArray = replaceSecret(event.getArgumentArray());
-
-                return new LoggingEvent(
+                AtomicReference<Instant> customTimestamp = new AtomicReference<>();
```

Comment:
Using entry.getKeyValuePairs.entrySet().stream().anyMatch() would avoid using an AtomicReference but it's not important so you can keep it as it.